src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -347,23 +344,13 @@ public bool TryWriteBytes(Span<byte> destination, out int bytesWritten)
         private void WriteIPv6Bytes(Span<byte> destination)
         {
             ushort[]? numbers = _numbers;
-            Debug.Assert(numbers != null && numbers.Length == NumberOfLabels);
+            Debug.Assert(numbers is { Length: NumberOfLabels });
 
-            if (BitConverter.IsLittleEndian)
+            if (BitConverter.IsLittleEndian && Vector128.IsHardwareAccelerated)
             {
-                if (Vector128.IsHardwareAccelerated)
-                {
-                    Vector128<ushort> ushorts = Vector128.LoadUnsafe(ref MemoryMarshal.GetArrayDataReference(numbers));
-                    ushorts = Vector128.ShiftLeft(ushorts, 8) | Vector128.ShiftRightLogical(ushorts, 8);
-                    ushorts.AsByte().StoreUnsafe(ref MemoryMarshal.GetReference(destination));
-                }
-                else
-                {
-                    for (int i = 0; i < numbers.Length; i++)
-                    {
-                        BinaryPrimitives.WriteUInt16BigEndian(destination.Slice(i * 2), numbers[i]);
```

Comment:
Isn't this fallback still needed if `Vector128` isn't accelerated?
(or replacing the `CopyTo` in the `else` block with this)

Reply:
Oops, indeed, reverted